### PR TITLE
picongpu --version

### PR DESCRIPTION
**Pull request by ax3l**
Supersedes PR #867, Output version information and exit with zero return state.

### Without MPI Init

```bash
$ ./bin/picongpu --version
```
```
PIConGPU: 0.4.0-dev
  Build-Type: Release

Third party:
  OS:         Linux-4.4.0-38-generic
  arch:       x86_64
  CXX:        GNU (4.9.2)
  CMake:      3.7.2 (Unix Makefiles)
  CUDA:       8000
  mallocMC:   2.2.0
  Boost:      106200
  MPI:        
    standard: 3.0
    flavor:   OpenMPI (1.8.6)
  PNGwriter:  0.5.6
  libSplash:  1.6.0 (Format 4.0)
  ADIOS:      1.10.0
```
(returns with 0 exit status)

### During Run

```bash
$ mpirun -n 2 ./bin/picongpu --versionOnce -g 64 32 32 --periodic 1 1 1 -d 2 1 1 -s 10
```
```
PIConGPU: 0.4.0-dev
  Build-Type: Release

Third party:
  OS:         Linux-4.4.0-38-generic
  arch:       x86_64
  CXX:        GNU (4.9.2)
  CMake:      3.7.2 (Unix Makefiles)
  CUDA:       8000
  mallocMC:   2.2.0
  Boost:      106200
  MPI:        
    standard: 3.0
    flavor:   OpenMPI (1.8.6)
  PNGwriter:  0.5.6
  libSplash:  1.6.0 (Format 4.0)
  ADIOS:      1.10.0
PIConGPUVerbose PHYSICS(1) | Sliding Window is OFF
PIConGPUVerbose PHYSICS(1) | Courant c*dt <= 1.00229 ? 1
PIConGPUVerbose PHYSICS(1) | species e: omega_p * dt <= 0.1 ? 0.0247974
PIConGPUVerbose PHYSICS(1) | y-cells per wavelength: 18.0587
PIConGPUVerbose PHYSICS(1) | macro particles per gpu: 65536
PIConGPUVerbose PHYSICS(1) | typical macro particle weighting: 6955.06
PIConGPUVerbose PHYSICS(1) | UNIT_SPEED 2.99792e+08
PIConGPUVerbose PHYSICS(1) | UNIT_TIME 1.39e-16
PIConGPUVerbose PHYSICS(1) | UNIT_LENGTH 4.16712e-08
PIConGPUVerbose PHYSICS(1) | UNIT_MASS 6.33563e-27
PIConGPUVerbose PHYSICS(1) | UNIT_CHARGE 1.11432e-15
PIConGPUVerbose PHYSICS(1) | UNIT_EFIELD 1.22627e+13
PIConGPUVerbose PHYSICS(1) | UNIT_BFIELD 40903.8
PIConGPUVerbose PHYSICS(1) | UNIT_ENERGY 5.69418e-10
initialization time:  6sec 370msec = 6 sec
  0 % =        0 | time elapsed:                    0msec | avg time per step:   0msec
 10 % =        1 | time elapsed:                   15msec | avg time per step:  15msec
 20 % =        2 | time elapsed:                   29msec | avg time per step:  13msec
 30 % =        3 | time elapsed:                   43msec | avg time per step:  14msec
 40 % =        4 | time elapsed:                   57msec | avg time per step:  14msec
 50 % =        5 | time elapsed:                   70msec | avg time per step:  13msec
 60 % =        6 | time elapsed:                   84msec | avg time per step:  14msec
 70 % =        7 | time elapsed:                   99msec | avg time per step:  14msec
 80 % =        8 | time elapsed:                  113msec | avg time per step:  14msec
 90 % =        9 | time elapsed:                  126msec | avg time per step:  13msec
100 % =       10 | time elapsed:                  140msec | avg time per step:  13msec
calculation  simulation time: 140msec = 0 sec
full simulation time:  6sec 553msec = 6 sec
```
(only printed once by rank zero and continues normal execution)

### CODE REVIEW

File: docs/TBG_macros.cfg [hunk 1]
```diff
@@ -72,6 +72,10 @@ TBG_projectPath
 TBG_dstPath
 
 
+# version information on startup
+TBG_verson="--versionOnce"
```
Comment (psychocoderHPC) [inline docs/TBG_macros.cfg]: a `i` is missing should be `TBG_version`

**Comment by PrometheusPi** [comment]
Would it not be easier to use if `--version` would only be printed from rank 0 if mpi is used instead of creating a second command line flag `--versionOnce`?

**Comment by psychocoderHPC** [comment]
`--version` exiting the program after the version is printed and `--versionOnce` is printing the version number and keep the simulation alive.

**Comment by ax3l** [comment]
Yes, but --version is standard Unix behavior and --versionOnce is just something verbose that we need for regular runs reproducibility, so we created two flags.

Using tbg won't see those flags anyway, since they are in our default cfgs.

**Pull request merged**